docs/developer/notification-center.md

```diff
@@ -95,3 +97,49 @@ This has the following fields:
 |key|ID of the user preference|
 |value|Value to store in the user preference when the corresponding notification is marked as read|
 |unsetValue|Value to store in the user preference when the corresponding notification is marked as unread (optional - when not specified, the preference will be set to an empty string)|
+
+# Adding Notifications in a UI Extension
+
+One common use case is to add/manage notifications at load time in a UI Extension.
+
+To do this, you can leverage the `onReady` UI Extension hook that is invoke when the user logs in and the notifications have been loaded and initialized.
+
+This can be set in the `init` function of your UI Extension with the `setOnReady` function, for example:
+
+```
+import { IPlugin } from '@shell/core/types';
+
+// Init the package
+export default function(plugin: IPlugin) {
+
+  // ... Other initialization code ...
+
+  // Set the `onReady` function to allow us to add a notification at log in
+  plugin.setOnReady(async (store: any) => {
+    await store.dispatch('notifications/add', { <notification> });
```

Comment:
Yes, I think it should suffice this.

Reply:
Oh wait, still one issue actually.
If we have no store we have no translation, so we should expose in somehow also i18n or try to translate within the notification component.

<img width="654" alt="Screenshot 2025-06-25 at 21 51 22" src="https://github.com/user-attachments/assets/cdb43ed7-4220-4363-89d5-2648a7d808ed" />

Reply:
You should be able to move the notification declaration into the `setOnReady` and access `t` via the store. I will make it easier to access `t`.